File: Misc/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
       
       int largestProfit = 0;
       int low = Integer.MAX_VALUE;

       for (int i = 0; i < prices.length; i ++) {
    		int price = prices[i];
    		if (price < low) {
    			low = price;
    		}
    		int currentProfit = price - low; 
    		largestProfit = Math.max(largestProfit, currentProfit);

       } 
       return largestProfit;
    }
}

File: Sliding Window/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        max_profit = 0

        # Lowest price we've found so far
        lowest_so_far = float('inf')

        for price in prices:

            lowest_so_far = min(lowest_so_far, price)

            # The most profit we can get with this current price is the difference between the current price 
            # and the lowest price we've found so far before it.
            current_max_profit = price - lowest_so_far

            # Update overall max profit if the current profit is greater than any profit we found so far.
            max_profit = max(max_profit, current_max_profit)

        return max_profit